src/ProductSync.php

```diff
@@ -507,97 +495,61 @@ public static function trigger_async_feed_registration_asap() {
-
-		if ( $feed && 'success' === $feed['status'] && isset( $feed['data']->location_config->full_feed_fetch_location ) ) {
-			self::log( 'Added merchant feed: ' . $feed_args['feed_location'] );
-			return $feed['data']->id;
-		}
-
-		return false;
-	}
-
-
 	/**
 	 * Handles feed registration using the given arguments.
 	 * Will try to create a merchant if none exists.
 	 * Also if a different feed is registered, it will update using the URL in the
 	 * $feed_args.
 	 *
-	 * @param array $feed_args The arguments used to create the feed.
-	 *
 	 * @return boolean|string
 	 *
 	 * @throws \Exception PHP Exception.
 	 */
-	private static function register_feed( $feed_args ) {
+	private static function register_feed() {
 
 		// Get merchant object.
-		$merchant   = Merchants::get_merchant( $feed_args );
+		$merchant   = Merchants::get_merchant();
 		$registered = false;
 
 		if ( ! empty( $merchant['data']->id ) && 'declined' === $merchant['data']->product_pin_approval_status ) {
+
 			$registered = false;
 			self::log( 'Pinterest returned a Declined status for product_pin_approval_status' );
-		} elseif ( ! empty( $merchant['data']->id ) && ! isset( $merchant['data']->product_pin_feed_profile ) ) {
-			// No feed registered, but we got a merchant.
-			$registered = self::do_add_merchant_feed( $merchant['data']->id, $feed_args );
-		} elseif ( $feed_args['feed_location'] === $merchant['data']->product_pin_feed_profile->location_config->full_feed_fetch_location ) {
-			// Feed registered.
-			$registered = $merchant['data']->product_pin_feed_profile->id;
-			self::log( 'Feed registered for merchant: ' . $feed_args['feed_location'] );
+
 		} else {
-			$product_pin_feed_profile    = $merchant['data']->product_pin_feed_profile;
-			$product_pin_feed_profile_id = false;
-			$prev_registered             = self::get_registered_feed_id();
-			if ( false !== $prev_registered ) {
-				try {
-					$feed                        = API\Base::get_merchant_feed_report( $merchant['data']->id, $prev_registered );
-					$product_pin_feed_profile_id = $feed['data']->feed_profile_id;
-				} catch ( \Throwable $e ) {
-					$product_pin_feed_profile_id = false;
-				}
-			}
 
-			if ( false === $product_pin_feed_profile_id ) {
-				$configured_path = dirname( $product_pin_feed_profile->location_config->full_feed_fetch_location );
-				$local_path      = dirname( $feed_args['feed_location'] );
+			// Check if there is a registered feed id.
+			$prev_registered = self::get_registered_feed_id();
+
+			// Update feed if there was no one registered previously.
+			if ( ! $prev_registered ) {
 
-				if ( $configured_path === $local_path && $feed_args['country'] === $product_pin_feed_profile->country && $feed_args['locale'] === $product_pin_feed_profile->locale ) {
-					// We can assume we're on the same site.
+				// The response only contains the merchant id.
+				$response = Merchants::update_or_create_merchant();
```

Comment:
Do we need to run this again here since its already run in the `get_merchant()` in line [511](https://github.com/saucal/pinterest-for-woocommerce/pull/7/files#diff-9eb8ac4928cebc9cc9e52da272589aafa0747ad026d7f101c0f9d6b9951af91fR511) ?

As I understand, especially on a new account `::update_or_create_merchant();` will be called in `Merchants::get_merchant();` and also again here. What's the case we are trying to catch ?